File: test/rql_test/importRethinkDB.js
#!/usr/bin/env node

var path = require('path');
var fs = require('fs');

// -- load rethinkdb from the proper location

rethinkdbLocation = process.env.JAVASCRIPT_DRIVER_DIR || path.join(__dirname, '..', '..', 'build', 'packages', 'js');
if (fs.existsSync(path.resolve(rethinkdbLocation, 'rethinkdb.js')) == false) {
    process.stdout.write('Could not locate the javascript drivers at the expected location: ' + rethinkdbLocation);
    process.exit(1);
}
var r        = require(path.resolve(rethinkdbLocation, 'rethinkdb'));
var protodef = require(path.resolve(rethinkdbLocation, 'proto-def'));
var net      = require(path.resolve(rethinkdbLocation, 'net'));

if (require.main === module) {
    banner = 'The RethinkDB driver has been imported as `r`, ';
    
    // setup default conn object
    var DRIVER_PORT = process.env.RDB_DRIVER_PORT || net.DEFAULT_PORT;
    var DRIVER_HOST = process.env.RDB_SERVER_HOST || net.DEFAULT_HOST;
    var conn = null;
    r.connect({'host':DRIVER_HOST, 'port':DRIVER_PORT})
    .then(function(c) {
        conn = c;
        banner += ' and a connection to ' + DRIVER_HOST + ':' + DRIVER_PORT + ' is available as `conn`.';
    }).error(function(err) {
        banner += ' but a default connection to ' + DRIVER_HOST + ':' + DRIVER_PORT + ' could not be made.';
    }).then(function() {
        // print the banner
        console.log(banner)
        
        // start the repl
        var repl = require('repl');
        var cli  = repl.start({ replMode: repl.REPL_MODE_MAGIC });
        cli.context.r = r;
        cli.context.rethindb = r;
        cli.context.protodef = protodef;
        if (conn) {
            cli.context.conn = conn;
        }
    });
    
} else {
    console.log('Using RethinkDB client from: ' + rethinkdbLocation)
    module.exports.r = r;
    module.exports.protodef = protodef;
}
    

File: test/rql_test/importRethinkDB.rb
#!/usr/local/bin/ruby
# Copyright 2014 RethinkDB, all rights reserved.

# -- get the location of the driver

rethinkdbDriverPath = File.expand_path(ENV['RUBY_DRIVER_DIR'] || File.join(File.dirname(__FILE__), '..', '..', 'build', 'drivers', 'ruby', 'lib'))

# -- import the RethinkDB driver

if !Dir.exists?(rethinkdbDriverPath) or !File.exists?(File.join(rethinkdbDriverPath, 'rethinkdb.rb'))
  abort "Unable to locate the Ruby driver from " + rethinkdbDriverPath
end
if !File.exists?(File.join(rethinkdbDriverPath, 'ql2.pb.rb'))
  abort "Ruby driver is not built: " + rethinkdbDriverPath
end

require File.join(rethinkdbDriverPath, 'rethinkdb.rb')
include RethinkDB::Shortcuts

# -- validate the location

if rethinkdbDriverPath != File.dirname(r.method(:connect).source_location[0])
  abort "Wrong Ruby driver loaded, expected from: " + rethinkdbDriverPath + " but got :" + File.dirname(r.method(:connect).source_location[0])
end
